AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Container/ContainerUIHostingController.swift

```
@@ -55,12 +55,14 @@ class ContainerUIHostingController: UIHostingController<ContainerUIHostingContro
             .receive(on: RunLoop.main)
             .removeDuplicates()
             .sink(receiveValue: { orientation in
-                if orientation == .portrait || orientation == .landscape {
+                let portrait = orientation == .portrait
+                let landscape = orientation == .landscapeLeft || orientation == .landscapeRight
```

Wouldn't orientation `.landscape` include `.landscapleLeft` and `.landscapeRight`?

---

The orientation `.landscape`  would represent the landscape orientation "in general". However in this scenario the app would need to know the current version of the landscape mode (left vs right) in order not to fall back to the incorrect one.

---

Since then, `if orientation == .portrait || orientation == .landscape { }` should also work, correct? Because `.landscape` includes `landscapeLeft` and `landscapeRight`

---

@AmyL219 ,  I think if we use `orientation == .landscape` here, the rotation would not work for the views conform to `LockPhoneOrientation`, mainly because the supportedOrientation for `LockPhoneOrientation` for was set to either .landscapeLeft and landscapeRight.

The if clause for `orientation == .landscape` would return false for either `.landscapeRight` or `.landscapeLeft`. We'd need the exact left/right orientation in `LockPhoneOrientation` to make sure the screen orientation won't be accidentally changed.